### ISSUE TITLE
Support `super` and `self` in find_definition

### ISSUE THREAD

**Issue by jakerr** [phildawes/racer#189]
```rust
pub fn iamhere() { println!("here"); }

mod inner { pub use super::iamhere; }
//                         ^ find-definition with cursor here returns nothing.

fn main() {
   iamhere();
// ^ find-definition with cursor here works.
   self::iamhere();
//       ^ find-definition with cursor here also does *not* work.
}
```

The lookup should work for inner mods in the same file as well as mods other files included in their parent with
```rust
mod inner;
```

**Comment by phildawes** [comment]
Hi Jake, thanks for filing this. Marking as bug

**Comment by phildawes** [comment]
I added a hack to support for self and super at the start of paths. I'm not sure it's the best way to do it yet but for now this fixes the bug. thanks @jakerr

**Comment by jakerr** [comment]
Awesome thanks for that quick fix! I verified that it's working after 264288383adabffc2469cfebd1ece842627bd561 for all the cases I reported.
